src/main.rs

```diff
@@ -0,0 +1,251 @@
+use anyhow::{anyhow, Context, Result};
+use base64::{engine::general_purpose, Engine as _};
+use clap::{Parser, Subcommand};
+use josekit::jwe::alg::direct::DirectJweAlgorithm::Dir;
+use josekit::jwk::Jwk;
+use serde::{Deserialize, Serialize};
+use std::io::{self, Read, Write};
+use std::process::Command as StdCommand;
+use std::thread;
+use std::time::Duration;
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Config {
+    url: Vec<String>,
+    resource_repository: String,
+    resource_type: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct ClevisHeader {
+    pin: String,
+    url: Vec<String>,
+    resource_repository: String,
+    resource_type: String,
+    resource_tag: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Key {
+    pub key_type: String,
+    pub key: String,
+}
+
+fn fetch_and_prepare_jwk(
+    urls: &[String],
+    resource_repository: &str,
+    resource_type: &str,
+    resource_tag: &str,
+) -> Result<Jwk> {
```

Comment:
That's fine. Alternatively it can be a single 'path' string of  repository/type/tag